CHANGELOG: update changelog for Packer 1.11.0 (#13001)
hashicorp/packer: CHANGELOG.md

### IMPROVEMENTS:
* core: Bump github.com/hashicorp/hcp-sdk-go from 0.90.0 to 0.96.0.
     [GH-12935](https://github.com/hashicorp/packer/pull/12935)
     [GH-12942](https://github.com/hashicorp/packer/pull/12942)
     [GH-12960](https://github.com/hashicorp/packer/pull/12960)
     [GH-12979](https://github.com/hashicorp/packer/pull/12979)
* core: Bump github.com/hashicorp/packer-plugin-sdk from 0.5.2 to 0.5.3
     [GH-12932](https://github.com/hashicorp/packer/pull/12932)

> * .github: updated actions in preparation for Node 16 deprecation

Reviewer: We generally don't include these in the CHANGELOG because they can add a lot of noise. I generally treat them as tech-debt.

Author: Yeah sounds good to me, I wasn't sure if we should include them, but leaving them out is probably for the best as users don't really care about maintenance of the repo (nor should they)

     [GH-12959](https://github.com/hashicorp/packer/pull/12959)
     [GH-12958](https://github.com/hashicorp/packer/pull/12958)
     [GH-12957](https://github.com/hashicorp/packer/pull/12957)
     [GH-12956](https://github.com/hashicorp/packer/pull/12956)
* .github: Bump go version to 1.12.10
     [GH-12984](https://github.com/hashicorp/packer/pull/12984)
* core: Bump go-getter/v2 from 2.2.1 to v2.2.2
     [GH-12988](https://github.com/hashicorp/packer/pull/12988)
* datasource/http: don't error on 2xx code
     Previous versions of Packer only supported 200 as a success case for the http
     datasource. This change makes any status code from 200 to 299 successful.
     [GH-12989](https://github.com/hashicorp/packer/pull/12989)
* hcl2: add textencodebase64 and textdecodebase64 funcs
     For feature parity with Terraform, and since having access to strings
     encoded in something that is not UTF-8 is required in some cases, we add
     both the textencodebase64 and textdecodebase64 functions to HCL2 templates.
     Please note these functions return base64 encoded byte slices because of how
     cty/hcl defines strings (NFC normalised, UTF-8 encoded).

>      [GH-12997](https://github.com/hashicorp/packer/pull/12997)

Reviewer: Let's move this under FEATURES.

     1.0.0-dev). Users are encouraged to notify plugin maintainers of any
     version mismatches.
     [GH-12915](https://github.com/hashicorp/packer/pull/12915), [GH-12953](https://github.com/hashicorp/packer/pull/12953), [GH-12972](https://github.com/hashicorp/packer/pull/12972)
* core: don't load plugins with metadata in name
     To avoid confusion with multiple plugins that report the same effective version,
     plugins installed need to have no metadata in their name.

>      When installed through packer commands, the metadata is scrubbed from the name of the

Reviewer: ```suggestion
     When installed through Packer commands, the metadata is scrubbed from the name of the
```

* core/hcp: fix potential race condition when storing plugin details to the HCP
     Packer metadata storage map.
     [GH-12936](https://github.com/hashicorp/packer/pull/12936)

> * core: fix listing on windows with .exe ext

Reviewer: ```suggestion
* core: fix  plugin listing on Windows
```

>      This bug fix addresses some bugs present in the alpha releases of 1.11, where
>      while revising the loading routines for Packer, we broke discovery for certain
>      commands on Windows builds.
>      [GH-12981](https://github.com/hashicorp/packer/pull/12981)

Reviewer: ```suggestion
     This fix addresses bugs present in the alpha releases of 1.11, where
     the discovery of Windows binaries were not matching against the filename extension (.ext).
     [GH-12981](https://github.com/hashicorp/packer/pull/12981)
```
